portfolio/src/main/webapp/script.js

```diff
@@ -336,11 +336,55 @@ function buildCommentFragment(comment) {
  * @return {!DocumentFragment} The populated comment fragment.
  */
 function populateCommentFragment(fragment, comment) {
-  fragment.querySelector(".body").append(comment);
+  const date = new Date(comment.timestamp);
+
+  fragment.querySelector('.body').append(comment.content);
+  fragment.querySelector('.date').append(date.toLocaleString());
+  fragment.querySelector('.title').style.setProperty(
+    '--indicator-color', 
+    getSentimentColor(comment.sentimentScore, comment.sentimentMagnitude)
+  );
 
   return fragment;
 }
 
+/**
+ * Calculates a color between red and green, based on the sentiment
+ * data provided. If the sentiment score is -1, the color will be
+ * fully red. If it is 1, the color will be fully green. Values
+ * in between these will result in intermediate colors. The brightness
+ * of the color will be determined by the sentiment magnitude.
+ * 
+ * @param {number} sentimentScore Number in range [-1, 1], which will determine
+ *    the hue of the color.
+ * @param {number} sentimentMagnitude Number in range [0, 1], which will determine
+ *    the lightness of the color
+ * 
+ * @return {string} A string containing a CSS color
+ */
+function getSentimentColor(sentimentScore, sentimentMagnitude) {
+  const hue = mapRange(sentimentScore, -1, 1, 0, 120);
```

Comment:
can you add comments to these arguments that are a bit ambiguous?

Reply:
Done!